api/share/utils.py

```diff
@@ -69,83 +74,100 @@ def _enqueue_update_share(osfresource):
         enqueue_task(async_update_resource_share.s(_osfguid_value))
 
 
-@celery_app.task(bind=True, max_retries=4, acks_late=True)
-def task__update_share(self, guid: str, is_backfill=False):
+@celery_app.task(
+    bind=True,
+    acks_late=True,
+    max_retries=4,
+    retry_backoff=True,
+)
+def task__update_share(self, guid: str, is_backfill=False, osfmap_partition_name='MAIN'):
     """
-    This function updates share  takes Preprints, Projects and Registrations.
-    :param self:
-    :param guid:
-    :return:
+    Send SHARE/trove current metadata record(s) for the osf-guid-identified object
     """
-    resp = _do_update_share(guid, is_backfill=is_backfill)
+    _osfmap_partition = OsfmapPartition[osfmap_partition_name]
+    _osfid_instance = apps.get_model('osf.Guid').load(guid)
+    if _osfid_instance is None:
+        raise ValueError(f'unknown osfguid "{guid}"')
+    _resource = _osfid_instance.referent
+    _is_deletion = _should_delete_indexcard(_resource)
+    _response = (
+        pls_delete_trove_record(_resource, osfmap_partition=_osfmap_partition)
+        if _is_deletion
+        else pls_send_trove_record(
+            _resource,
+            is_backfill=is_backfill,
+            osfmap_partition=_osfmap_partition,
+        )
+    )
     try:
-        resp.raise_for_status()
+        _response.raise_for_status()
     except Exception as e:
-        if self.request.retries == self.max_retries:
-            log_exception(e)
-        elif resp.status_code >= 500:
-            try:
-                self.retry(
-                    exc=e,
-                    countdown=(random.random() + 1) * min(60 + settings.CELERY_RETRY_BACKOFF_BASE ** self.request.retries, 60 * 10),
+        log_exception(e)
+        if HTTPStatus(_response.status_code).is_server_error:
+            raise self.retry(exc=e)
+    else:  # success response
+        if not _is_deletion:
+            # enqueue followup task for supplementary metadata
+            _next_partition = _next_osfmap_partition(_osfmap_partition)
```

Comment:
as it is, yes, `MAIN` should be updated first, because shtrove will ignore supplementary updates when there's no main index-card to supplement, or if that card is marked deleted (this is... less than ideal, but shouldn't be a problem while chaining updates like this -- could be a future improvement)

that said, the two supplements could be sent in parallel just fine...